### ISSUE TITLE
Select multiple - search in items : error value.match is not a function (due to SafeHtmlImpl)

### ISSUE THREAD

**Issue by jcs13** [valor-software/ng2-select#438]
Hello

In "select multiple", when you search an items (input text), I've an error : 

> EXCEPTION: Error in ./SelectComponent class SelectComponent - inline template:106:19 caused by: t.match is not a function

(Chrome & FF + Angular Final 2.0.0)
Same issue with your demo.

The error is into the HighlightPipe :  `var tagList = value.match(tagRE);` due to value is not a string but instance of SafeHtmlImpl

I don't known if it's the best soltuion, but you could fix it with : 

```
        if (value instanceof SafeHtmlImpl) {
            value = value.changingThisBreaksApplicationSecurity;
        }
```

thkx

**Comment by alpeshv** [comment]
Not sure how you will get hold of SafeHtmlImpl. But I am using it locally with the following fix.

```
public transform(safeHtml:any, args:any[]):any {            
    var value = safeHtml.changingThisBreaksApplicationSecurity;
    ...
}
```

**Comment by theunreal** [comment]
Same as https://github.com/valor-software/ng2-select/issues/437

**Comment by theunreal** [comment]
@alpeshv Where did you place /replace this line with?

**Comment by alpeshv** [comment]
@theunreal Inside select-pipes.ts, changing signature of transform method and adding one line inside it at the top. Rest will remain same.

**Comment by Scipionh** [comment]
I can't highlight enough, how important it is to fix this asap.

**Comment by GuillaumeDK57** [comment]
+1
Typing filter throws exception : 

### myExceptionHandler.service.ts:15 Stack : TypeError: value.match is not a function

```
at HighlightPipe.transform (http://127.0.0.1:9000/assets/lib/ng2-select/components/select/select-pipes.js:15:33)
at eval (http://127.0.0.1:9000/assets/lib/angular__core/src/linker/view_utils.js!transpiled:120:26)
at DebugAppView._View_SelectComponent8.detectChangesInternal (SelectComponent.ngfactory.js:821:136)
at DebugAppView.AppView.detectChanges (http://127.0.0.1:9000/assets/lib/angular__core/src/linker/view.js!transpiled:334:26)
at DebugAppView.detectChanges (http://127.0.0.1:9000/assets/lib/angular__core/src/linker/view.js!transpiled:439:56)
at DebugAppView.AppView.detectContentChildrenChanges (http://127.0.0.1:9000/assets/lib/angular__core/src/linker/view.js!transpiled:352:31)
at DebugAppView._View_SelectComponent7.detectChangesInternal (SelectComponent.ngfactory.js:745:8)
at DebugAppView.AppView.detectChanges (http://127.0.0.1:9000/assets/lib/angular__core/src/linker/view.js!transpiled:334:26)
at DebugAppView.detectChanges (http://127.0.0.1:9000/assets/lib/angular__core/src/linker/view.js!transpiled:439:56)
at DebugAppView.AppView.detectContentChildrenChanges (http://127.0.0.1:9000/assets/lib/angular__core/src/linker/view.js!transpiled:352:31)
```

**Comment by theunreal** [comment]
@jcs13 where did you place this:
```
if (value instanceof SafeHtmlImpl) {
            value = value.changingThisBreaksApplicationSecurity;
        }
```

**Comment by theunreal** [comment]
@alpeshv Changing select-pipes.d.ts with the following code:

```
import { PipeTransform } from '@angular/core';
export declare class HighlightPipe implements PipeTransform {
    public transform(safeHtml:any, args:any[]):any {            
    var value = safeHtml.changingThisBreaksApplicationSecurity;
    }
}
export declare function stripTags(input: string): string;
```

don't update anything in the app, only when updating the JS files. I guess I need to re-build the package, how I can do that?

**Comment by alpeshv** [comment]
@theunreal I have not tried to change npm package directly inside my project yet but you shouldn't be changing them anyway. You should download and include select.module.ts in your project and all the related .ts files which are in the 'select' folder instead of downloading it using npm and change references accordingly. If your project is a ts project then your ts files are being transpiled into js files using tsc command somewhere. Anyway select2 is not stable at all. idField and textField properties are also not fully working. I had to fix them myself.

**Comment by Scipionh** [comment]
Any news on this ?

**Comment by alexjeen** [comment]
+1 having the same problem

**Comment by jppellerin** [comment]
+1

**Comment by faizan-ali** [comment]
+1

**Comment by yobud** [comment]
+1

**Comment by alexjeen** [comment]
This is taking too long, it's blocking a project of ours. Can somebody make a PR to fix this? Maybe @alpeshv as you've already seemed to fix this?

**Comment by alpeshv** [comment]
@alexjeen I don't have permission to make a PR on this repo unfortunately. Anyway there are so many PRs pending. There are also issues with id and value fields. You will be better off copying the code and make changes locally. They have done nice work upfront but allowed bugs being merged recently.

**Comment by alexjeen** [comment]
We've switched to a jQuery implementation so it's not needed anymore.

**Comment by jcs13** [comment]
FIX with 1.1.1 : 
select: issue #437 (#454) (926e14d)

Thanks :) !
